### PR TITLE
Pass relevant override options for Container's detached/attach flow

### PR DESCRIPTION
**Pull request by kian-thompson**
## Description

There is currently a gap in functionality when using the `load` flow vs the `createDetached` flow on the Container. For the `createDetached` flow, there was no way to provide override values for things like `canReconnect` and `clientDetailsOverride` on the `IContainerCreateProps` interface.

Now, the Loader's `createDetachedContainer(...)` and `rehydrateDetachedContainerFromSnapshot(...)` methods contain an optional props argument to pass in these values.

Also, for the `attach` method there was no way to configure connecting to the deltaConnect (normally done through the `LoaderHeader.loadMode` header in the `load` flow). Now, the Container's `attach(...)` method contains an optional props argument to pass in this values.

[AB#4958](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4958)

### CODE REVIEW

File: packages/loader/container-loader/src/loader.ts [hunk 1]
```diff
@@ -357,8 +357,19 @@ export class Loader implements IHostLoader {
 		return this;
 	}
 
-	public async createDetachedContainer(codeDetails: IFluidCodeDetails): Promise<IContainer> {
-		const container = await Container.createDetached(this.services, codeDetails);
+	public async createDetachedContainer(
+		codeDetails: IFluidCodeDetails,
+		containerCreateHeaders?: IRequestHeader,
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/loader.ts:362]: lets avoid using header here, as we are trying to move away from request like patterns, i would just take an object, something like:
```suggestion
		props?: {
		canReconnect?: boolean,
		clientDetailsOverride?: IClientDetails
		}
```

File: packages/loader/container-loader/src/loader.ts [hunk 2]
```diff
@@ -357,8 +357,19 @@ export class Loader implements IHostLoader {
...
+	): Promise<IContainer> {
+		// "this.services" doesn't contain property for clientDetailsOverride
+		const container = await Container.createDetached(
+			{
+				canReconnect: containerCreateHeaders?.[LoaderHeader.reconnect],
+				clientDetailsOverride: containerCreateHeaders?.[LoaderHeader.clientDetails],
+				...this.services,
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/loader.ts:369]: this then gets simplified to 
```suggestion
				...this.props,
				...this.services,
```

File: packages/loader/container-loader/src/container.ts [hunk 3]
```diff
@@ -2453,6 +2446,34 @@ export class Container
 			this.runtime.setConnectionState(state && !readonly, this.clientId);
 		}
 	}
+
+	private handleDeltaConnectionArg(
+		connectionArgs: IConnectionArgs,
+		deltaConnectionArg?: "none" | "delayed",
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/container.ts:2452]: would it make sense to add these properties to IConnectionArgs

Comment (kian-thompson) [reply packages/loader/container-loader/src/container.ts:2452]: I don't think so? The `IConnectionArgs` are for the `DeltaManager` and this `deltaConnectionArg` isn't really relevant to the DM, so it probably shouldn't be coupled to the DM.

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**